src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -118,31 +116,27 @@ public ValueTask<FlushResult> WriteDataToPipeAsync(ReadOnlySpan<byte> buffer, Ca
 
         public ValueTask<FlushResult> WriteStreamSuffixAsync()
         {
+            ValueTask<FlushResult> result = default;
```

Comment:
⚠️ The semantics of this method changed for the _second_ caller. Prior to this change, the following sequence could occur:

1. Thread 1 calls WriteStreamSuffixAsync
1. A task is returned to thread 1 is not complete
1. Thread 2 calls WriteStreamSuffixAsync
1. The task return to thread 1 in step (2) is completed
1. A task is returned to thread 2 which _is_ complete

Starting with this change, it is likely that step (5) will occur _before_ step (4).

Reply:
> The semantics of this method changed for the second caller

Yes, hence my comment to Stephen that it's the change he and I spoke about:
https://github.com/dotnet/aspnetcore/pull/31221/files/9c6ea2dba9b4b8c64c314dae84ccfdad5063a981#r601005357

And the change in behavior is nowhere near as complicated as you outline: previously the second caller would get the same task, now they get the default.  There shouldn't be a second caller, though.

Reply:
Yep. Basically this means if someone calls `HttpResponse.BodyWrite.CompleteAsync()` and doesn't await it, Kestrel won't observe output backpressure before either closing the connection or attempting to read the next request. I'm not too worried.